### ISSUE TITLE
Constant Timeout Errors

### ISSUE THREAD

**Issue by jonathanmwood** [nitaybz/homebridge-onkyo-pioneer#5]
I’m getting constant timeout errors. It seems to come and go and I’m not sure why. Relevant section of log below:

> [2/9/2021, 11:17:48 AM] [OnkyoPioneer] AV Receiver - Getting State
> [2/9/2021, 11:17:52 AM] [OnkyoPioneer] SubError [TimeoutError]: Timeout when sending: !1PWRQSTN
>     at afterTimeout (/homebridge/node_modules/homebridge-onkyo-pioneer/node_modules/bluebird/js/release/timers.js:49:15)
>     at Timeout.timeoutTimeout [as _onTimeout] (/homebridge/node_modules/homebridge-onkyo-pioneer/node_modules/bluebird/js/release/timers.js:76:13)
>     at listOnTimeout (internal/timers.js:554:17)
>     at processTimers (internal/timers.js:497:7)
> [2/9/2021, 11:17:52 AM] [OnkyoPioneer] Could NOT get state from "AV Receiver" : Timeout when sending: !1PWRQSTN
> [2/9/2021, 11:17:52 AM] [OnkyoPioneer] AV Receiver - found cached state in storage
> [2/9/2021, 11:17:52 AM] [OnkyoPioneer] ERROR OCCURRED: Timeout when sending: !1PWRQSTN

**Comment by nitaybz** [comment]
I'm working on a fix...

**Comment by nitaybz** [comment]
Latest version v0.0.18 should fix the issue

**Comment by mwtremblay** [comment]
Hi, this seems to have reared its ugly head again.

[4/12/2022, 9:19:12 PM] [homebridge-onkyo-pioneer] ERROR OCCURRED: Timeout when sending: !1PWRQSTN
[4/12/2022, 9:19:12 PM] [homebridge-onkyo-pioneer] Could NOT get state from "Onkyo" : Timeout when sending: !1PWRQSTN

Plugin Version: v0.0.18
Homebridge: v1.4.0, Alpine Linux Docker image

**Comment by awallace916** [comment]
I have been getting this timeout error lately as well.

**Comment by awallace916** [comment]
TLDR: Unplug your receiver, wait 30 seconds, plug it back in. Problem fixed.

Just wanted to post this in case it helps anyone else. I've noticed from time to time that my receiver will quit connecting to my network. At first I thought it was because I was running a wireless connection so I wired it into my network but would still deal with the receiver failing to connect to my network. I would notice this when trying to connect to it via the Onkyo app on my phone. I had a "duh" moment when I realized that I might be getting these "Timeout..." errors in homebridge. Sure enough, when I ran a network test through the receiver it failed to connect to my network. The fix for me has been to unplug the receiver, wait 30 seconds, plug it back in. It does seem to take a minute to initialize the network once it turns back on so give it a good 60 seconds before running the network test again. Hope this helps!

**Comment by nitaybz** [comment]
Should be addressed in v0.1.0 (just published). Per-field state reads, log-once for unsupported commands, and HB v2 compat are all in. Please update and let me know if anything still misbehaves.

Sorry for the long delay on this.
